RecoMuon/MuonIdentification/plugins/ME0SegmentMatcher.cc

```diff
@@ -125,193 +73,161 @@ ME0SegmentMatcher::ME0SegmentMatcher(const edm::ParameterSet& pas) {
-    std::vector<ME0Muon> TempStore; 
+  Handle<ME0SegmentCollection> OurSegments;
+  ev.getByToken(OurSegmentsToken_,OurSegments);
 
-    Handle <TrackCollection > generalTracks;
-    //ev.getByLabel <TrackCollection> ("generalTracks", generalTracks);
-    ev.getByToken(generalTracksToken_,generalTracks);
+  auto oc = std::make_unique<std::vector<ME0Muon>>(); 
+  std::vector<ME0Muon> TempStore; 
 
+  Handle <TrackCollection > generalTracks;
+  ev.getByToken(generalTracksToken_,generalTracks);
 
-    int TrackNumber = 0;
+  int TrackNumber = 0;
     
-    for (std::vector<Track>::const_iterator thisTrack = generalTracks->begin();
-	 thisTrack != generalTracks->end(); ++thisTrack,++TrackNumber){
-      //Initializing our plane
-
-      //Remove later
-      if (std::abs(thisTrack->eta()) < 1.8) continue;
-
-      float zSign = thisTrack->pz() > 0 ? 1.0f : -1.0f;
-
-      const float zValue = 526.75 * zSign;
-
-      Plane *plane = new Plane(Surface::PositionType(0,0,zValue),Surface::RotationType());
+  for (std::vector<Track>::const_iterator thisTrack = generalTracks->begin();
+       thisTrack != generalTracks->end(); ++thisTrack,++TrackNumber){
+    //Initializing our plane
 
-      //Getting the initial variables for propagation
+    //Remove later
+    if (std::abs(thisTrack->eta()) < 1.8) continue;
 
-      int chargeReco = thisTrack->charge(); 
-      GlobalVector p3reco, r3reco;
+    //Getting the initial variables for propagation
+    float zSign = thisTrack->pz() > 0 ? 1.0f : -1.0f;
 
-      p3reco = GlobalVector(thisTrack->outerPx(), thisTrack->outerPy(), thisTrack->outerPz());
-      r3reco = GlobalVector(thisTrack->outerX(), thisTrack->outerY(), thisTrack->outerZ());
+    ReferenceCountingPointer<Plane> plane = Plane::build(Surface::PositionType(0,0,zSign*526.75),Surface::RotationType());
 
-      AlgebraicSymMatrix66 covReco;
-      //This is to fill the cov matrix correctly
-      AlgebraicSymMatrix55 covReco_curv;
-      covReco_curv = thisTrack->outerStateCovariance();
-      FreeTrajectoryState initrecostate = getFTS(p3reco, r3reco, chargeReco, covReco_curv, &*bField);
-      getFromFTS(initrecostate, p3reco, r3reco, chargeReco, covReco);
+    int chargeReco = thisTrack->charge(); 
+    GlobalVector p3reco, r3reco;
 
-      //Now we propagate and get the propagated variables from the propagated state
-      //SteppingHelixStateInfo startrecostate(initrecostate);
-      //SteppingHelixStateInfo lastrecostate;
-      TrajectoryStateOnSurface lastrecostate;
+    p3reco = GlobalVector(thisTrack->outerPx(), thisTrack->outerPy(), thisTrack->outerPz());
+    r3reco = GlobalVector(thisTrack->outerX(), thisTrack->outerY(), thisTrack->outerZ());
 
-      //const SteppingHelixPropagator* ThisshProp = 
-      //dynamic_cast<const SteppingHelixPropagator*>(&*shProp);
+    AlgebraicSymMatrix66 covReco;
+    //This is to fill the cov matrix correctly
+    AlgebraicSymMatrix55 covReco_curv;
+    covReco_curv = thisTrack->outerStateCovariance();
```

Comment:
hi @jshlee - can this be a const AlgebraicSymMatrix55 & instead?

Reply:
done